packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -451,104 +450,154 @@ export class DataStores implements IDisposable {
 		requestHeaderData: RuntimeHeaderData,
 	): Promise<FluidDataStoreContext> {
 		const headerData = { ...defaultRuntimeHeaderData, ...requestHeaderData };
-		const request = { url: id };
-
-		this.validateNotDeleted(id, request, headerData);
+		this.validateNotDeleted(id, headerData);
 
 		const context = await this.contexts.getBoundOrRemoted(id, headerData.wait);
 		if (context === undefined) {
 			// The requested data store does not exits. Throw a 404 response exception.
+			const request: IRequest = {
+				url: id,
+				headers: requestHeaderData,
```

Comment:
Why are you adding this?  It's a little suspect (IIRC) because headers is supposed to have string values but `RuntimeHeaderData` is the parsed/strongly typed data